### PR TITLE
fix(android): Ensure the MultiPostprocessor cache key is set correctly

### PR DESCRIPTION
**Pull request by matt-oakes**
## Summary

If the `MultiPostprocessor` wrapped postprocessors which returned a `null` cache key, it would return a `MultiCacheKey` with `mull` elements in the list. When compared these would match in the cache, causing issues where cached values were used where they shouldn't be.

This was causing issues when an `Image` component was using `resizeMode="repeat"` once, and then drawing another component with the same image source but with a different size. Because the source matched and the MultiPostprocessor cache key was incorrectly matching, it would return the repeated image with the wrong size, causing issues.

With this fix cache is now correctly implemented.

## Changelog

[Android] [Fixed] - Ensure the MultiPostprocessor cache key is set correctly to avoid issues when redisplaying images with a different size with resizeMode="repeat"

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/views/image/MultiPostprocessor.java [hunk 1]
```diff
@@ -52,9 +52,20 @@ public String getName() {
   public CacheKey getPostprocessorCacheKey() {
```
Comment (mdvacca) [inline ReactAndroid/src/main/java/com/facebook/react/views/image/MultiPostprocessor.java]: have you check what are the side effects of this method returning null?

File: ReactAndroid/src/main/java/com/facebook/react/views/image/MultiPostprocessor.java [hunk 2]
```diff
@@ -52,9 +52,20 @@ public String getName() {
...
     LinkedList<CacheKey> keys = new LinkedList<>();
     for (Postprocessor p : mPostprocessors) {
-      keys.push(p.getPostprocessorCacheKey());
+      CacheKey cacheKey = p.getPostprocessorCacheKey();
+      if (cacheKey != null) {
+        keys.push(cacheKey);
+      }
+    }
+
+    switch (postprocessors.size()) {
```
Comment (mdvacca) [inline ReactAndroid/src/main/java/com/facebook/react/views/image/MultiPostprocessor.java]: can you double check this method?
where is postprocessors defined? maybe you need to use keys instead?

File: ReactAndroid/src/main/java/com/facebook/react/views/image/MultiPostprocessor.java (hunk 1 above)
Comment (mdvacca) [inline ReactAndroid/src/main/java/com/facebook/react/views/image/MultiPostprocessor.java]: can you add the @Nullable annotation?

Comment (matt-oakes) [reply ReactAndroid/src/main/java/com/facebook/react/views/image/MultiPostprocessor.java]: Looking at the uses inside Fresco, `null` is an accepted and expected value:

https://github.com/facebook/fresco/search?p=2&q=getPostprocessorCacheKey&unscoped_q=getPostprocessorCacheKey

**Comment by matt-oakes** [comment]
@mdvacca Thanks for the review. I've made those changes now

**Pull request closed without merging**